components/automate-ui/src/app/modules/desktop/daily-check-in/daily-check-in.component.spec.ts

```diff
@@ -0,0 +1,26 @@
+import { CUSTOM_ELEMENTS_SCHEMA } from '@angular/core';
```

Comment:
I added spec files and the scaffolding to start testing for all the new desktop components I touched, but ended up not adding tests for the tooltips as I couldn't find a good way to trigger hover in unit tests.  When I thought about it more, this seems ok, because the tooltip text is not dynamic, and chef-tooltip already has tests for displaying the tip text.

Reply:
Does the 'hover' action need to be unit tested here? Should be sufficient to just check for the presence of the 'info' icon and that it has a tooltip presently associated with it.

"Does `#icon` element exist?"
"Does tooltip `[for="icon"]` element exist?"

Reply:
Checking for existence of the icon element is a good idea - it seems counter to the idea of testing behavior though. Thoughts? Existence tests seem better than no tests though.

Reply:
Tradeoffs. You can test the hover behavior here but are you going to test every tooltip hover interaction everywhere? It'd be thorough and could possibly catch some unforeseen things but would be really slow.

We should test hover behavior in `chef-tooltip`'s test suite but everywhere else it should be sufficient enough to say that if I use the element as documented, it should work as documented. If you're doing something unique with the tooltip interaction or styling on a particular page, sure, add extra tests there but otherwise you should be fine.

For example, we don't usually write tests to ensure that `input` elements with a `disabled` attribute cannot not have their `value` changed. We usually just check that the `disabled` attribute is present and assume the element will behave correctly.

Reply:
appreciate the input, i'm gonna add some of existence tests here before merging 👍

Reply:
My notions are very much inline with @scottopherson's, a unit test should be testing things relevant to the current "unit"--generally the current component--and you assume other components perform as advertised.

The one thing I might add is that you can expand the thinking so rather than "existence tests" you could think of them as "value tests" to perhaps be a bit more palatable: _Does attribute `x` have value `y`?_  The case of a "raw" attribute--like `disabled`--is the degenerate case, where the value is just implicitly true.

Reply:
Thank you! I appreciate the responses 👍